src/overlays/actors/ovl_Boss_01/z_boss_01.c

```diff
@@ -246,191 +647,2935 @@ ActorInit Boss_01_InitVars = {
+        Math_ApproachZeroF(&this->actor.scale.y, 1.0f, 0.00075f);
+        Math_ApproachF(&this->bugDrawDmgEffScale, 1.0f, 0.5f, 0.15f);
+
+        if (this->actor.scale.y < 0.001f) {
+            Math_ApproachZeroF(&this->bugDrawDmgEffAlpha, 1.0f, 0.05f);
+            if (this->bugDrawDmgEffAlpha < 0.01f) {
+                Actor_Kill(&this->actor);
+                Item_DropCollectibleRandom(play, NULL, &this->actor.world.pos, 0x60);
+                sBugCount--;
+            }
+        }
+
+        Math_ApproachS(&this->actor.shape.rot.x, 0, 1, 0x800);
+    } else {
+        this->actor.shape.rot.x -= 0x2000;
+        this->actor.shape.rot.z += 0x1000;
+    }
+}
+
+void Boss01_Bug_UpdateDamage(Boss01* this, PlayState* play) {
+    Vec3f additionalVelocity;
+    s32 pad[2];
+    u8 damage;
+    ColliderInfo* acHitInfo;
+    OdolwaEffect* effect = play->specialEffects;
+
+    if (this->bugACCollider.base.acFlags & AC_HIT) {
+        this->bugACCollider.base.acFlags &= ~AC_HIT;
+        acHitInfo = this->bugACCollider.info.acHitInfo;
+
+        if (this->damageTimer == 0) {
+            Matrix_RotateYS(this->actor.yawTowardsPlayer, MTXMODE_NEW);
+            if (acHitInfo->toucher.dmgFlags & 0x300000) {
+                this->damageTimer = 10;
+                Matrix_MultVecZ(-10.0f, &additionalVelocity);
+                this->additionalVelocityX = additionalVelocity.x;
+                this->additionalVelocityZ = additionalVelocity.z;
+            } else {
+                this->damageTimer = 15;
+                this->damageFlashTimer = 15;
+                Matrix_MultVecZ(-20.0f, &additionalVelocity);
+                this->additionalVelocityX = additionalVelocity.x;
+                this->additionalVelocityZ = additionalVelocity.z;
+                if (this->actor.colChkInfo.damageEffect == BUG_DMGEFF_STUN) {
+                    Boss01_Bug_SetupStunned(this, play);
+                    Actor_PlaySfx(&this->actor, NA_SE_EN_COMMON_FREEZE);
+                } else {
+                    damage = this->actor.colChkInfo.damage;
+                    this->actor.colChkInfo.health -= damage;
+                    Boss01_Bug_SetupDamaged(this, play);
+                    Actor_PlaySfx(&this->actor, NA_SE_EN_MIZUBABA2_DAMAGE);
+                }
+            }
+        }
+    }
+
+    // If Odolwa dies, instantly kill any bugs in the room too. Setting their y-velocity to 0 will skip the part of
+    // their death where they fly backwards; they will immediately shrink and burst into a single blue flame.
+    if ((sOdolwa != NULL) && (sOdolwa->actionFunc == Boss01_DeathCutscene)) {
+        Boss01_Bug_SetupDead(this, play);
+        this->actor.velocity.y = 0.0f;
+        this->actor.speed = 0.0f;
+    }
+
+    if ((effect->type == ODOLWA_EFFECT_RING_OF_FIRE) && (effect->timer < 150)) {
+        f32 sqrt = sqrtf(SQ(effect->pos.x - this->actor.world.pos.x) + SQ(effect->pos.z - this->actor.world.pos.z));
```

Comment:
```suggestion
        f32 distXZ = sqrtf(SQ(effect->pos.x - this->actor.world.pos.x) + SQ(effect->pos.z - this->actor.world.pos.z));
```
Maybe name it like this?